src/common/baseTelemetryAppender.ts

```diff
@@ -47,6 +56,40 @@ export class BaseTelemetryAppender implements ITelemetryAppender {
 		this._telemetryClient.logEvent(eventName, data);
 	}
 
+	/**
+	 * **DANGEROUS**: Logs an event regardless of telemetry level.
+	 * This should only be used in controlled environments i.e. CI pipelines
+	 * @param eventName The named of the event to log
+	 * @param data The data contained in the event
+	 */
+	logEventDangerously(eventName: string, data?: AppenderData): void {
+		if (!this._telemetryClient) {
```

Comment:
_telemetryClient isn't set until it finishes being instantiated, so this will miss events while the client is being created. The same check should be used as in the other logEvent method

Reply:
Thanks I was trying to be clever. We actually don't need this dangerous logs so I've removed them. Also fixed #94